data/blog/how-to-refactor-and-not-break-things.md

```diff
@@ -0,0 +1,101 @@
+---
+title: 'How to Refactor and Not Break Things'
+date: '2024-04-30'
+tags: ['python', 'refactoring', 'sdk']
+draft: false
+summary: 'How we completed a huge refactoring of a software used by thousands of developers without breaking things.'
+images: ['/images/how-to-refactor-and-not-break-things/hero.jpg']
+layout: PostLayout
+authors: ['antonpirker']
+---
+
+In our Python SDK, we completed a huge refactoring, and I want to write down how we pulled this off without breaking (almost) anything and how we managed to stay mostly backward compatible.
+
+## The Initial Situation
+
+When you add the Sentry SDK to your project it hooks into the execution of your application and collects data during execution of your application. Depending on the frameworks or libraries you use the data is collected in different parts of your project and is sent to Sentry in other parts. 
+If your project consists of multiple services (like the frontend, the backend, some microservices, or worker processes consuming items from a queue) the Sentry SDK also propagates tracing information between those services. This makes it possible to link the data from all your services into one trace.
```

Comment:
Should we call it application? or system? instead of project? Not sure if it's just me but, "Sentry Project" is a Sentry thing, but when I think of my projects, I don't necessarily think of an app, or a larger system, or product. 

```suggestion
If your application consists of multiple services (like frontend, backend, some microservices, or worker processes consuming items from a queue) the Sentry SDK also propagates tracing information between those services. This makes it possible to link the data from all your services into one trace.
```